File: src/ast/NestedName.java
package ast;

import java.io.StringWriter;

public class NestedName extends ParentNode {
    private BaseNode _name;

    public NestedName(BaseNode name, BaseNode type) {
        super(NodeType.NestedName, type);
        _name = name;
    }

    @Override
    public String GetName() {
        return _name.GetName();
    }

    @Override
    public void PrintLeft(StringWriter writer) {
        child.Print(writer);
        writer.write("::");
        _name.Print(writer);
    }
}

File: src/de/fearlesstobi/demangler/ast/NestedName.kt
package de.fearlesstobi.demangler.ast

import java.io.StringWriter

class NestedName(val name: BaseNode, type: BaseNode?) : ParentNode(NodeType.NestedName, type) {
    override fun getName(): String? {
        return name.getName()
    }

    public override fun printLeft(writer: StringWriter) {
        child!!.print(writer)
        writer.write("::")
        name.print(writer)
    }

}